data/dal.go

```diff
@@ -46,5 +48,9 @@ func (dal *DAL) Ping() error {
 
 // Close closes the connection to the database.
 func (dal *DAL) Close() error {
-	return dal.db.Close()
+	database, ok := dal.db.(*pg.DB)
+	if !ok {
+		return errors.New("dal.db not of type *pg.DB")
+	}
+	return database.Close()
```

Comment:
This is fine, but maybe we should just panic if this doesn't work. I say that because usually you do checks like this when you've received something and you're not quite sure what it is. But in this case an error would mean that we've fundamentally broken our DB integration so a panic might be appropriate. Up to you, let me know what you think.

Reply:
That's a good point, I'll make it panic

![](https://media0.giphy.com/media/HUkOv6BNWc1HO/giphy.gif)

Reply:
@bfbachmann actually apparently `.Close()` is never called anywhere in the codebase 😅 

And come to think of it, in the event `.Close()` is called, Rocket should be in the process of shutting down already - it might be fine to leave this as is. I'll try to find a place to put a `.Close()` somewhere.